add blog for newly added service (#4857)
webdriverio/webdriverio: website/blog/2019-12-05-wiremock.md

---

> title: WireMock as a service

Reviewer: ```suggestion
title: WireMock as a Service
```

author: Erwin Heitzman
authorURL: http://github.com/erwinheitzman
authorImageURL: https://avatars1.githubusercontent.com/u/15839059?s=80&v=4
---

> # wdio-wiremock-service

Reviewer: ```suggestion
# The Wiremock Service
```

We are pleased to announce that we now have a new mocking service for WebdriverIO called `wdio-wiremock-service`.

## What can it do?

> This service helps you to run WireMock seamlessly when running tests with [WebdriverIO](https://webdriver.io). It uses the well known [Maven](https://mvnrepository.com/repos/central) repository to download the WireMock jar for you which is then automatically installed, started and stopped. Stay up to date by joining the community over at [Gitter](https://gitter.im/erwinheitzman/wdio-wiremock-service) to join others and for support.

Reviewer: ```suggestion
This service helps you to run WireMock seamlessly when running tests with [WebdriverIO](https://webdriver.io). It uses the well known [Maven](https://mvnrepository.com/repos/central) repository to download the WireMock jar for you which is then automatically installed, started and stopped. Stay up to date by joining the community over at [Gitter](https://gitter.im/erwinheitzman/wdio-wiremock-service) for help and support.
```

A few things you can do with this service:

> - Automatically run WireMock alongside the WebdriverIO testrunner

Reviewer: ```suggestion
- Automatically run [WireMock](http://wiremock.org/) alongside the WebdriverIO testrunner
```

- Match request URLs, methods, headers cookies and bodies using a wide variety of strategies. First class support for JSON and XML.
- Configure WireMock with all the available [options](https://github.com/erwinheitzman/wdio-wiremock-service/blob/master/README.md#options)

## Installation

```bash
npm install wdio-wiremock-service --save-dev
```

Instructions on how to install `WebdriverIO` can be found [here.](https://webdriver.io/docs/gettingstarted.html)

## Configuration

In order to use the service you need to add it to your service array:

```js
// wdio.conf.js
export.config = {
  // ...
  services: ['wiremock'],
  // ...
};
```

There are a lot more options that are available for configuration, for a full overview go [here](https://github.com/erwinheitzman/wdio-wiremock-service/blob/master/README.md#options).

## Writing tests

> *wdio-wiremock-service* creates a directory (`./mock` by default) where creates two directories (`mappings` and `__files`). It will use the `mappings` directory to look for mock data.

Reviewer 2: should that be `where it creates`? Also, i suggest changing 'two directories' to 'two subdirectories' to be clearer (assuming that's true)

Reviewer: Maybe a sentence would be nice for what the `__files` directory is for if you mention it.
